collect_app/src/main/java/org/odk/collect/android/widgets/utilities/GeoDataRequester.java

```diff
@@ -0,0 +1,123 @@
+package org.odk.collect.android.widgets.utilities;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.location.Location;
+import android.os.Bundle;
+
+import org.javarosa.core.model.FormIndex;
+import org.javarosa.core.model.QuestionDef;
+import org.odk.collect.android.R;
+import org.odk.collect.android.listeners.PermissionListener;
+import org.odk.collect.android.utilities.PermissionUtils;
+import org.odk.collect.android.widgets.interfaces.GeoButtonClickListener;
+
+import java.text.DecimalFormat;
+
+import timber.log.Timber;
+
+public class GeoDataRequester implements GeoButtonClickListener {
+    public static final String LOCATION = "gp";
+    public static final String ACCURACY_THRESHOLD = "accuracyThreshold";
+    public static final String READ_ONLY = "readOnly";
+    public static final String DRAGGABLE_ONLY = "draggable";
+    public static final double DEFAULT_LOCATION_ACCURACY = 5.0;
+
+    @Override
+    public void requestGeoIntent(Context context, FormIndex index, PermissionUtils permissionUtils,
+                                 WaitingForDataRegistry waitingForDataRegistry, Class activityClass, Bundle bundle, int requestCode) {
+        permissionUtils.requestLocationPermissions((Activity) context, new PermissionListener() {
```

Comment:
`permissionUtils` could probably be a field of this class and come in at the constructor rather than be passed in as a method argument.